fabric/src/test/java/TestStyledText.java

```diff
@@ -522,4 +522,19 @@ public void styledText_getStringWithNonChatFormattingColors() {
                 styledText.getString(PartStyle.StyleType.DEFAULT),
                 "StyledText.getString() returned an unexpected value.");
     }
+
+    @Test
+    public void styledText_inheritesHoverEvents() {
+        final Component component = Component.empty()
+                .withStyle(style ->
+                        style.withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.literal("hover"))))
+                .append(Component.literal("inherited hover effect"));
+
+        StyledText styledText = StyledText.fromComponent(component);
+
+        Assertions.assertEquals(
```

Comment:
Just want to check: without the fix, does the test fail?

Reply:
Yes. That is what causes tue linked issue. I suspected it after  I looked at it, and it was easy to confirm.